### ISSUE TITLE
Focusable img elements should fail if they have an empty alt

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2583]
I think it would be good if axe-core distinguished between focusable images and non-focusable ones. For example, I think the following should fail:

```html
<img alt="" tabindex="0" />
```

To do this, we'd need to split up the image-alt rule. Maybe exclude focusable images from it and have those tested in a separate img-alt-focusable rule?
